### PR TITLE
Optimize DeviceModel code.

### PR DESCRIPTION
**Pull request by robabram**
* Swtiched from QtEnum to Python IntEnum type.
* Renamed property names and variable names to match Enum.
* Used zip() function to combine two lists into DeviceRolePropertyMap dictionary.

### CODE REVIEW

File: ultimarc/devices/mini_pac.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def _create_message_(self, config_file, cur_device_config=None):
                         pass
 
                 except KeyError:
-                    _logger.debug(_(f'Pin {pin.name} does not exists in Mini-pac device'))
```
Comment (katie-snow) [inline ultimarc/devices/mini_pac.py:380]: this name variable shouldn't be changed to product_name.

File: ultimarc/devices/mini_pac.py [hunk 2]
```diff
@@ -365,7 +365,7 @@ def _create_message_(self, config_file, cur_device_config=None):
                         elif alternate_action in macro_dict:
                             data.bytes[alternate_action_index] = macro_dict[alternate_action]
                         elif alternate_action:
-                            _logger.info(_(f'{pin.name} alternate action "{alternate_action}" is not a valid value'))
```
Comment (katie-snow) [inline ultimarc/devices/mini_pac.py:368]: this name variable shouldn't be changed to product_name.

File: ultimarc/devices/mini_pac.py [hunk 3]
```diff
@@ -349,14 +349,14 @@ def _create_message_(self, config_file, cur_device_config=None):
...
                 try:
-                    action_index, alternate_action_index, shift_index = PinMapping[pin.name]
+                    action_index, alternate_action_index, shift_index = PinMapping[pin.product_name]
                     action = pin.action.upper()
                     if action in IPACSeriesMapping:
                         data.bytes[action_index] = IPACSeriesMapping[action]
                     elif action in macro_dict:
                         data.bytes[action_index] = macro_dict[action]
                     else:
-                        _logger.info(_(f'{pin.name} action "{action}" is not a valid value'))
```
Comment (katie-snow) [inline ultimarc/devices/mini_pac.py:359]: this name variable shouldn't be changed to product_name.

File: ultimarc/devices/mini_pac.py [hunk 4]
```diff
@@ -349,14 +349,14 @@ def _create_message_(self, config_file, cur_device_config=None):
             # 0x40 in the shift position for all pins designated as shift pins in json config
             for pin in config.pins:
                 try:
-                    action_index, alternate_action_index, shift_index = PinMapping[pin.name]
```
Comment (katie-snow) [inline ultimarc/devices/mini_pac.py:352]: this name variable shouldn't be changed to product_name.

File: ultimarc/devices/mini_pac.py [hunk 5]
```diff
@@ -325,7 +325,7 @@ def _create_message_(self, config_file, cur_device_config=None):
                     if len(macro.action) > 0:
                         # Set the start point of the new macro
                         data.bytes[cur_position] = cur_macro
-                        macro_dict[macro.name.upper()] = cur_macro
```
Comment (katie-snow) [inline ultimarc/devices/mini_pac.py:328]: this name variable shouldn't be changed to product_name.

**Review by katie-snow — CHANGES_REQUESTED** [review]

File: ultimarc/devices/mini_pac.py (hunk 1 above)
Comment (robabram) [reply ultimarc/devices/mini_pac.py]: Done.

File: ultimarc/devices/mini_pac.py (hunk 2 above)
Comment (robabram) [reply ultimarc/devices/mini_pac.py]: Fixed.

File: ultimarc/devices/mini_pac.py (hunk 3 above)
Comment (robabram) [reply ultimarc/devices/mini_pac.py]: Fixed

File: ultimarc/devices/mini_pac.py (hunk 4 above)
Comment (robabram) [reply ultimarc/devices/mini_pac.py]: Fixed.

File: ultimarc/devices/mini_pac.py (hunk 5 above)
Comment (robabram) [reply ultimarc/devices/mini_pac.py]: Fixed.

**Review by katie-snow — APPROVED** [review]

**Pull request merged**